[Index management] Clean all tests after execution and navigate directly to the tab

SoniaSanzV opened the pull request:
## Summary
Space time project. I'm trying to improve the test for Kibana Management. In this PR, I did two things:
* Clean up all created resources after execution. This suite contained a lot of resources that wasn't deleted after execution. That complicates local testing and leads to failures if you re-run the tests. 
* Navigates directly to the desired tab when setting up the tests. This is not an anti-patter, but it was time consuming. We were doing this:
```
      await pageObjects.common.navigateToApp('indexManagement');
      // Navigate to the data streams tab
      await pageObjects.indexManagement.changeTabs('data_streamsTab');
      await pageObjects.header.waitUntilLoadingHasFinished();
```

But the `navigateToApp` method already contains a path parameter, so doing it this way, we can get rid of unnecessaries `waitUntilLoadingHasFinished` calls.
With this simple change, the execution time of serverless tests locally dropped from 2 minutes to 1 minute.

* Also, changes the data-test-subj for creating a component template from `createPipelineButton` to `createComponentTemplateButton`, that wasn't reflecting what the button really does.

Flay test runner stateful tests: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9177
Flay test runner serverless tests: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/9182



<!--ONMERGE {"backportTargets":["8.18","8.19","9.0","9.1"]} ONMERGE-->

<!--ONMERGE {"backportTargets":["8.19","9.1"]} ONMERGE-->

SoniaSanzV committed:
Clean all tests after execution and navigate directly to the tab

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/330714)
* Commit: 2dc1357dfd53880501108a8b6cdd78f9c022847f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
> ✅ unchanged

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330714","number":330714,"commit":"2dc1357dfd53880501108a8b6cdd78f9c022847f"}],"number":330714}
buildkite-pr-comment-->

SoniaSanzV committed:
Add changes to serverless

SoniaSanzV committed:
Add missing navigateTo

SoniaSanzV committed:
Unify cleaning for enrich

SoniaSanzV committed:
Merge branch 'main' into kbn_mngmt_test/index_management_tests_cleanup

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/333599)
* Commit: c8bf852b5ff6945363c21cf4eb3f1658545edadf

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a7fbdb3e1080ae335125e2d2818b488c0d6724ac) | [after](https://github.com/elastic/kibana/commit/c8bf852b5ff6945363c21cf4eb3f1658545edadf) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 686.9KB | 686.9KB | +18.0B |



### History
* :green_heart: [Build #330832](https://buildkite.com/elastic/kibana-pull-request/builds/330832) succeeded 8fa330ea3534914ff0e51bfcefc09ab7fc8ba75e
* :green_heart: [Build #330714](https://buildkite.com/elastic/kibana-pull-request/builds/330714) succeeded 2dc1357dfd53880501108a8b6cdd78f9c022847f

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333599","number":333599,"commit":"c8bf852b5ff6945363c21cf4eb3f1658545edadf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330832","number":330832,"commit":"8fa330ea3534914ff0e51bfcefc09ab7fc8ba75e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330714","number":330714,"commit":"2dc1357dfd53880501108a8b6cdd78f9c022847f"}],"number":333599}
buildkite-pr-comment-->

ElenaStoeva commented on x-pack/platform/test/functional/apps/index_management/create_enrich_policy/create_enrich_policy.ts:
```diff
@@ -43,22 +43,19 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       } catch (e) {
-        log.debug('[Teardown error] Error deleting test policy');
-        throw e;
+        log.debug(`[Teardown error] Error deleting test index: ${e.message}`);
```
We could separate these into two `try`-`catch` blocks so that if one fails the other resource is still deleted, and the log message would be more helpful (i.e. right now, if the policy deletion failed, the logs will say that the index deletion failed).

ElenaStoeva commented on x-pack/platform/test/functional/apps/index_management/enrich_policies_tab/enrich_policies_tab.ts:
```diff
@@ -56,20 +56,16 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       } catch (e) {
-        log.debug('[Teardown error] Error deleting test policy');
-        throw e;
+        log.debug(`[Teardown error] Error deleting test policy: ${e.message}`);
```
Same comment here

ElenaStoeva commented on x-pack/platform/test/serverless/functional/test_suites/management/index_management/create_enrich_policy.ts:
```diff
@@ -58,10 +54,10 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       } catch (e) {
-        log.debug('[Teardown error] Error deleting test policy');
-        throw e;
+        log.debug(`[Teardown error] Error deleting test index: ${e.message}`);
```
Same comment here

ElenaStoeva commented on x-pack/platform/test/serverless/functional/test_suites/management/index_management/enrich_policies.ts:
```diff
@@ -52,21 +52,16 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
       } catch (e) {
-        log.debug('[Teardown error] Error deleting test policy');
-        throw e;
+        log.debug(`[Teardown error] Error deleting test index: ${e.message}`);
```
Same comment here

ElenaStoeva reviewed (commented):
Great job @SoniaSanzV - it's nice to see the test running time reduction! Changes lgtm with one suggestion for an improvement.

SoniaSanzV committed:
Separate index and policy deletion

SoniaSanzV commented:
> Great job @SoniaSanzV - it's nice to see the test running time reduction! Changes lgtm with one suggestion for an improvement.

Thank you for the review! I've addressed your suggestions in my last commit!

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/333775)
* Commit: d74139cc48ed3520a43abfbdfc0b2f34ba6adc2b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a7fbdb3e1080ae335125e2d2818b488c0d6724ac) | [after](https://github.com/elastic/kibana/commit/d74139cc48ed3520a43abfbdfc0b2f34ba6adc2b) | diff |
| --- | --- | --- | --- |
| `indexManagement` | 686.9KB | 686.9KB | +18.0B |



### History
* :green_heart: [Build #333599](https://buildkite.com/elastic/kibana-pull-request/builds/333599) succeeded c8bf852b5ff6945363c21cf4eb3f1658545edadf
* :green_heart: [Build #330832](https://buildkite.com/elastic/kibana-pull-request/builds/330832) succeeded 8fa330ea3534914ff0e51bfcefc09ab7fc8ba75e
* :green_heart: [Build #330714](https://buildkite.com/elastic/kibana-pull-request/builds/330714) succeeded 2dc1357dfd53880501108a8b6cdd78f9c022847f

cc @SoniaSanzV

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333775","number":333775,"commit":"d74139cc48ed3520a43abfbdfc0b2f34ba6adc2b"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/333599","number":333599,"commit":"c8bf852b5ff6945363c21cf4eb3f1658545edadf"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330832","number":330832,"commit":"8fa330ea3534914ff0e51bfcefc09ab7fc8ba75e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/330714","number":330714,"commit":"2dc1357dfd53880501108a8b6cdd78f9c022847f"}],"number":333775}
buildkite-pr-comment-->

ElenaStoeva reviewed (approved):
Lgtm, thanks for the last changes!

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Looks like this PR has backport PRs but they still haven't been merged. Please merge them ASAP to keep the branches relatively in sync.
cc: @SoniaSanzV

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 232198` locally
cc: @SoniaSanzV

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 232198` locally
cc: @SoniaSanzV

kibanamachine commented:
Friendly reminder: Looks like this PR hasn’t been backported yet.
To create automatically backports add a `backport:*` label or prevent reminders by adding the `backport:skip` label.
You can also create backports manually by running `node scripts/backport --pr 232198` locally
cc: @SoniaSanzV
